src/tests/tribol_proximity_check.cpp

```diff
@@ -0,0 +1,274 @@
+
+    // fixed options
+
+    // kinematic constant penalty stiffness
+    constexpr tribol::RealT penalty = 1.0;
+    // boundary element attributes of contact surface 1, the top surface of the bottom block
+    auto contact_surf_1 = std::set<int>( { 3 } );
+    // boundary element attributes of contact surface 2, the bottom surface of the top block
+    auto contact_surf_2 = std::set<int>( { 5 } );
+
+    // create a mesh with two blocks
+    // clang-format off
+    shared::ParMeshBuilder mesh( MPI_COMM_WORLD, shared::MeshBuilder::Merged( {
+      shared::MeshBuilder::SquareMesh( 1, 1 ),
+      shared::MeshBuilder::SquareMesh( 1, 1 )
+        .translate( { 0.0, 1.0 - mesh_interpenetration } )
+        .updateAttrib( 1, 2 )
+        .updateBdrAttrib( 1, 5 )
+        .updateBdrAttrib( 3, 6 )
+      } ) );
+    // clang-format on
+
+    // grid function for higher-order nodes
+    mesh.setNodesFEColl( mfem::H1_FECollection( order, dim ) );
+
+    SetUpCommonPlaneProblem( mesh, contact_surf_1, contact_surf_2, penalty, binning_proximity, binning_method );
+  }
+
+  std::tuple<shared::ParMeshBuilder, std::set<int>, std::set<int>> SetUp3DMesh()
+  {
+    constexpr int dim = 3;
+    // polynomial order of the finite element discretization
+    auto order = std::get<0>( GetParam() );
+    // mesh interpenetration (scaled by element radius)
+    auto mesh_interpenetration = std::get<2>( GetParam() ) * std::sqrt( 2 );
+
+    // fixed options
+
+    // boundary element attributes of contact surface 1, the top surface of the bottom block
+    auto contact_surf_1 = std::set<int>( { 6 } );
+    // boundary element attributes of contact surface 2, the bottom surface of the top block
+    auto contact_surf_2 = std::set<int>( { 7 } );
+
+    // create a mesh with two cubes
+    // clang-format off
+    shared::ParMeshBuilder mesh( MPI_COMM_WORLD, shared::MeshBuilder::Merged( {
+      shared::MeshBuilder::HypercubeMesh( dim, 1 ),
+      shared::MeshBuilder::HypercubeMesh( dim, 1 )
+        .translate( { 0.0, 0.0, 1.0 - mesh_interpenetration } )
+        .updateAttrib( 1, 2 )
+        .updateBdrAttrib( 1, 7 )
+        .updateBdrAttrib( 6, 8 )
+      } ) );
+    // clang-format on
+
+    // grid function for higher-order nodes
+    mesh.setNodesFEColl( mfem::H1_FECollection( order, dim ) );
+
+    return { std::move( mesh ), std::move( contact_surf_1 ), std::move( contact_surf_2 ) };
+  }
+
+  void SetUp3DCommonPlaneProblem( tribol::BinningMethod binning_method )
+  {
+    // binning proximity value
+    auto binning_proximity = std::get<1>( GetParam() );
+    // kinematic constant penalty stiffness
+    constexpr tribol::RealT penalty = 1.0;
+
+    auto mesh = SetUp3DMesh();
+
+    SetUpCommonPlaneProblem( std::get<0>( mesh ), std::get<1>( mesh ), std::get<2>( mesh ), penalty, binning_proximity,
+                             binning_method );
+  }
+
+  void SetUpMortarProblem( tribol::BinningMethod binning_method )
```

Comment:
Consider renaming consistent with how you rename SetupCommonPlaneProblem(). I realize there are no 2D and 3D variants for mortar. This routine kind of does it all. It Sets up the problem AND updates. I think I have something like this in TestMesh called SetupAndUpdate...

Reply:
Yeah, it's similar to the SetupAndUpdate routine, though for the Tribol MFEM API.  It might be good to break this out into a separate file too, though in a follow on PR.

Reply:
On second thought, I think I'll just get rename things for now using your suggestions.  If we decide on a better way to "drive" Tribol in this example, we can change it later.